src/main/java/frc/robot/auto_manager/AutoManager.java

```diff
@@ -190,33 +179,29 @@ private Command intakeAnyStepNotes(AutoNoteStep step) {
         .until(() -> robotManager.getState().hasNote);
   }
 
+  /**
+   * Depending on the robot's current position, follow a path towards areas where there are probably
+   * stray notes. If we see a note, end the command so we can start running cleanup. Otherwise, it
+   * will just complete the path and sit there.
+   */
   private Command searchForNoteForCleanupCommand() {
-    Translation2d robot = localization.getUsedPose().getTranslation();
-    if (robot.getDistance(getClosestSpeaker().getTranslation()) < 4.0) {
-
-      return Commands.sequence(
-              getSpeakerCleanupPath().stream()
-                  .map(
-                      pose -> {
-                        return AutoBuilder.pathfindToPose(pose, DEFAULT_CONSTRAINTS);
-                      })
-                  .toArray(Command[]::new))
-          .until(
-              () -> {
-                return noteTrackingManager.getNearestNotePoseRelative(robot, 5.0).isPresent();
-              });
-    }
-    return Commands.sequence(
-            getMidlineCleanupPath().stream()
-                .map(
-                    pose -> {
-                      return AutoBuilder.pathfindToPose(pose, DEFAULT_CONSTRAINTS);
-                    })
-                .toArray(Command[]::new))
-        .until(
-            () -> {
-              return noteTrackingManager.getNearestNotePoseRelative(robot, 5.0).isPresent();
-            });
+    // TODO: This possibly needs us to specifcy requirements or else something bad might happen
```

Comment:
refactored this command to be deferred, previously we were checking robot pose on command construction. which is bad vibes. in theory that's not a bug if we only construct the command while an auto is in progress. still though, not good practice to rely on that

i also noticed that the only thing changing depending on robot pose is the path to follow. so rather than repeat all the command code, i made it do the same thing either way just with a different path